test/core/end2end/tests/connectivity.cc

```diff
@@ -163,9 +169,74 @@ static void test_connectivity(grpc_end2end_test_config config) {
   cq_verifier_destroy(cqv);
 }
 
+static void cb_watch_connectivity(grpc_experimental_completion_queue_functor *functor, int success) {
+  callback_context* cb_ctx = (callback_context*) functor;
+
+  gpr_log(GPR_DEBUG, "cb_watch_connectivity called, verifying");
+
+  /* callback must not have errors */
+  GPR_ASSERT(success != 0);
+
+  gpr_mu_lock(&cb_ctx->mu);
+  cb_ctx->done = true;
+  gpr_mu_unlock(&cb_ctx->mu);
+}
+
+static void cb_shutdown(grpc_experimental_completion_queue_functor *functor, int success) {
+  gpr_log(GPR_DEBUG, "cb_shutdown called, nothing to do");
+}
+
+static void test_watch_connectivity_cq_callback(grpc_end2end_test_config config) {
+  grpc_experimental_completion_queue_functor shutdown_functor;
+  callback_context cb_ctx;
+  grpc_completion_queue * cq;
+  grpc_end2end_test_fixture f = config.create_fixture(nullptr, nullptr);
+
+  grpc_channel_args client_args;
+  client_args.args = nullptr;
+  client_args.num_args = 0;
+
+  config.init_client(&f, &client_args);
```

Comment:
I think you might be able to just pass `nullptr` for the channel args here, in which case we don't need `client_args` at all.